src/test/test-execute.c

```diff
@@ -623,13 +623,19 @@ static void test_exec_privatenetwork(Manager *m) {
 
 static void test_exec_oomscoreadjust(Manager *m) {
         test(m, "exec-oomscoreadjust-positive.service", 0, CLD_EXITED);
-        test(m, "exec-oomscoreadjust-negative.service", 0, CLD_EXITED);
+        if (detect_container() > 0)
```

Comment:
As I've just written in the other PR you created, `detect_container` is too broad a check. Plus, basically, these tests can fail because they don't have enough permissions to do what they're supposed to do and that can happen from containers too. The right approach would be to check available capabilities and the like, but I don't think it's worth it. I'd let them fail. After all, it's useful to know what works and what doesn't in the environment you're interested in.

Reply:
i agree with @evverx: this test should check caps and stuff